frontend/src/metabase/api/dataset.ts

```diff
@@ -1,9 +1,9 @@
 import { Api } from "metabase/api";
-import type { NativeQueryForm, StructuredQuery } from "metabase-types/api";
+import type { NativeQueryForm, DatasetQuery } from "metabase-types/api";
 
-export const nativeDatasetApi = Api.injectEndpoints({
+export const datasetApi = Api.injectEndpoints({
   endpoints: builder => ({
-    createNativeDataset: builder.mutation<NativeQueryForm, StructuredQuery>({
+    createNativeDataset: builder.mutation<NativeQueryForm, DatasetQuery>({
```

Comment:
Before this change
![image](https://github.com/metabase/metabase/assets/31325167/f9156061-51ac-412c-8313-774f973efc43)